src/videojs-contrib-hls.js

```diff
@@ -674,13 +674,20 @@ const HlsSourceHandler = function(mode) {
 
       let settings = videojs.mergeOptions(options, {hls: {mode}});
 
+      let previousBeforeRequest;
+
+      if (tech.hls && tech.hls.xhr) {
```

Comment:
On dispose is fine, but what about on source changes? Since the `hls` object is per player, shouldn't it continue to exist and not be disposed?